assets/app/scripts/services/data.js

```diff
@@ -159,6 +159,92 @@ angular.module('openshiftConsole')
     }
   	return deferred.promise;
   };
+  
+// type:      API type (e.g. "pods")
+// object:    API object data(eg. { kind: "Build", parameters: { ... } } )
+// context:   API context (e.g. {project: "..."})
+// opts:      http - options to pass to the inner $http call
+// Returns a promise resolved with response data or rejected with {data:..., status:..., headers:..., config:...} when the delete call completes.
+  DataService.prototype.create = function(type, object, context, opts) {
+  	opts = opts || {};
+  	var deferred = $q.defer();
+    if (context.projectPromise && type !== "projects") {
+      var self = this;
+      context.projectPromise.done(function(project) {
+        $http(angular.extend({
```

Comment:
We should clean all of these up, here is what an example might look like for the get method:

```
      var self = this;
      var fn = function() {
        $http(angular.extend({
          method: 'GET',
          url: self._urlForType(type, name, context, false, {namespace: context.projectName})
        }, opts.http || {}))
        .success(function(data, status, headerFunc, config, statusText) {
          deferred.resolve(data);
        })
        .error(function(data, status, headers, config) {
          deferred.reject({
            data: data,
            status: status,
            headers: headers,
            config: config
          });
        });
      };

      if (context.projectPromise && type !== "projects") {
        context.projectPromise.done(fn);
      }
      else {
        fn();
      }
```

If context.projectName exists on the context, it will use that as the namespace.  If you want to perform a request for something not in your current project, (like to get a template from another namespace) then you would pass a custom context object that explicitly defines namespace.  I have an example of this in catalog.js

```
DataService.list("templates", {namespace: "openshift"}, function(templates) { ......
```